clang/test/Driver/repo-toolchain-linux-musl.c

```diff
@@ -0,0 +1,215 @@
+// RUN:   | FileCheck -check-prefix=CHECK002 %s
+
+// CHECK002:       "-resource-dir" "[[RESOURCE_DIR:[^"]+]]"
+// CHECK002-NOT:   "/usr/local/musl/lib/crt1.t"
+// CHECK002-NOT:   "/usr/local/musl/lib/crt1_asm.t"
+// CHECK002-NOT:   "/usr/local/musl/lib/libc_repo.a"
+// CHECK002-NOT:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtbegin-x86_64.o"
+// CHECK002-NOT:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtend-x86_64.o"
+// CHECK002-NOT:   "-lclang_rt.builtins-x86_64"
+
+// -----------------------------------------------------------------------------
+// Checking the linked objects and libraries
+// Passing --musl -nostartfiles
+// -----------------------------------------------------------------------------
+// RUN: %clang -### -target x86_64-pc-linux-musl-repo -nostartfiles \
+// RUN:   -resource-dir=%S/Inputs/resource_dir \
+// RUN:   --sysroot=%S/Inputs/basic_linux_libcxx_tree \
+// RUN:   %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK003 %s
+
+// CHECK003:       "-resource-dir" "[[RESOURCE_DIR:[^"]+]]"
+// CHECK003-NOT:   {{.*}}basic_linux_libcxx_tree{{/|\\\\}}lib{{/|\\\\}}crt1.t
+// CHECK003-NOT:   {{.*}}basic_linux_libcxx_tree{{/|\\\\}}lib{{/|\\\\}}crt1_asm.t
+// CHECK003-NOT:   {{.*}}basic_linux_libcxx_tree{{/|\\\\}}lib{{/|\\\\}}libc_repo.a"
+// CHECK003-NOT:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtbegin-x86_64.o"
+// CHECK003-NOT:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtend-x86_64.o"
+// CHECK003:   "-lclang_rt.builtins-x86_64"
+
+// -----------------------------------------------------------------------------
+// Checking the linked objects and libraries
+// Passing --musl -nodefaultlibs
+// -----------------------------------------------------------------------------
+// RUN: %clang -### -target x86_64-pc-linux-musl-repo -nodefaultlibs \
+// RUN:   -resource-dir=%S/Inputs/resource_dir \
+// RUN:   --sysroot=%S/Inputs/basic_linux_libcxx_tree \
+// RUN:   %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK004 %s
+
+// CHECK004:   "-resource-dir" "[[RESOURCE_DIR:[^"]+]]"
+// CHECK004:   {{.*}}basic_linux_libcxx_tree{{/|\\\\}}lib{{/|\\\\}}crt1.t
+// CHECK004:   {{.*}}basic_linux_libcxx_tree{{/|\\\\}}lib{{/|\\\\}}crt1_asm.t
+// CHECK004:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtbegin-x86_64.o"
+// CHECK004:   "[[RESOURCE_DIR]]{{/|\\\\}}lib{{/|\\\\}}linux{{/|\\\\}}clang_rt.crtend-x86_64.o"
+// CHECK004-NOT:   -lclang_rt.builtins-x86_64
+
+// -----------------------------------------------------------------------------
+// Checking the linked objects and libraries
+// Not Passing -fno-use-init-array when musl is selected
+// -----------------------------------------------------------------------------
+// RUN: %clang -### -target x86_64-pc-linux-musl-repo %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHECK005 %s
+
+// RUN: %clang -### -target x86_64-pc-linux-gnu-repo %s 2>&1 \
```

Comment:
Weren't you going to move the *-gnu-repo tests out of the *-musl.c source file? They don't seem to fit here.

Reply:
Removed in the commit [0a629b99](https://github.com/SNSystems/llvm-project-prepo/pull/158/commits/0a629b99bb27274ce786c619db54e1a59d902947).